crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -0,0 +1,943 @@
+    pub fn parse_with_version(
+        bs: &[u8],
+        version: FormatVersion,
+        partition_type: &StructType,
+    ) -> Result<ManifestList, Error> {
+        match version {
+            FormatVersion::V2 => {
+                let schema = schema_to_avro_schema("manifest_list", &Self::v2_schema()).unwrap();
+                let reader = Reader::with_schema(&schema, bs)?;
+                let values = Value::Array(reader.collect::<Result<Vec<Value>, _>>()?);
+                from_value::<_serde::ManifestListV2>(&values)?.try_into(partition_type)
+            }
+            FormatVersion::V1 => {
+                let schema = schema_to_avro_schema("manifest_list", &Self::v1_schema()).unwrap();
+                let reader = Reader::with_schema(&schema, bs)?;
+                let values = Value::Array(reader.collect::<Result<Vec<Value>, _>>()?);
+                from_value::<_serde::ManifestListV1>(&values)?.try_into(partition_type)
+            }
+        }
+    }
+
+    /// Get the entries in the manifest list.
+    pub fn entries(&self) -> &[ManifestListEntry] {
+        &self.entries
+    }
+
+    /// Get the v2 schema of the manifest list entry.
+    pub(crate) fn v2_schema() -> Schema {
+        let fields = vec![
+            _schema::MANIFEST_PATH.clone(),
+            _schema::MANIFEST_LENGTH.clone(),
+            _schema::PARTITION_SPEC_ID.clone(),
+            _schema::CONTENT.clone(),
+            _schema::SEQUENCE_NUMBER.clone(),
+            _schema::MIN_SEQUENCE_NUMBER.clone(),
+            _schema::ADDED_SNAPSHOT_ID.clone(),
+            _schema::ADDED_FILES_COUNT_V2.clone(),
+            _schema::EXISTING_FILES_COUNT_V2.clone(),
+            _schema::DELETED_FILES_COUNT_V2.clone(),
+            _schema::ADDED_ROWS_COUNT_V2.clone(),
+            _schema::EXISTING_ROWS_COUNT_V2.clone(),
+            _schema::DELETED_ROWS_COUNT_V2.clone(),
+            _schema::PARTITIONS.clone(),
+            _schema::KEY_METADATA.clone(),
+        ];
+        Schema::builder().with_fields(fields).build().unwrap()
+    }
+
+    /// Get the v1 schema of the manifest list entry.
+    pub(crate) fn v1_schema() -> Schema {
+        let fields = vec![
+            _schema::MANIFEST_PATH.clone(),
+            _schema::MANIFEST_LENGTH.clone(),
+            _schema::PARTITION_SPEC_ID.clone(),
+            _schema::ADDED_SNAPSHOT_ID.clone(),
+            _schema::ADDED_FILES_COUNT_V1.clone().to_owned(),
+            _schema::EXISTING_FILES_COUNT_V1.clone(),
+            _schema::DELETED_FILES_COUNT_V1.clone(),
+            _schema::ADDED_ROWS_COUNT_V1.clone(),
+            _schema::EXISTING_ROWS_COUNT_V1.clone(),
+            _schema::DELETED_ROWS_COUNT_V1.clone(),
+            _schema::PARTITIONS.clone(),
+            _schema::KEY_METADATA.clone(),
+        ];
+        Schema::builder().with_fields(fields).build().unwrap()
+    }
+}
+
+/// This is a helper module that defines the schema field of the manifest list entry.
+mod _schema {
```

Comment:
The naming convention `_schema` seems a bit strange to me. What about using `fields` or `constant_fields` instead?